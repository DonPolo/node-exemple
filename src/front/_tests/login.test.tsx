import React from 'react';
import { mount } from 'enzyme';
import { BrowserRouter } from 'react-router-dom';
import Login from '../pages/login.page';
import fetch from './_mocks/fetch.mock';
import { SinonStub } from 'sinon';

describe('Test Login page', () => {
  let fakeFetch: SinonStub;
  beforeAll(() => {
    fakeFetch = fetch.mock();
  });
  test('Render the page correctly', async () => {
    const page = mount(
      <BrowserRouter>
        <Login />
      </BrowserRouter>,
    );
    expect(page).toMatchSnapshot();
  });
  afterAll(() => {
    fetch.unmock(fakeFetch);
  });
});
