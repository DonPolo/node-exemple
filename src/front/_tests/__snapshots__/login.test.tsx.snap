// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Login page Render the page correctly 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <LoginPage>
      <main>
        <form
          method="post"
          onSubmit={[Function]}
        >
          <h2>
            Log In
          </h2>
          <label
            className="label"
            htmlFor="pseudo"
          >
            Pseudo
            <input
              className="input"
              id="pseudo"
              name="pseudo"
              onChange={[Function]}
              type="text"
              value=""
            />
          </label>
          <label
            className="label"
            htmlFor="pwd"
          >
            Password
            <input
              className="input"
              id="pwd"
              name="pwd"
              onChange={[Function]}
              type="password"
              value=""
            />
          </label>
          <button
            className="btn"
            type="submit"
          >
            Continue
          </button>
        </form>
      </main>
    </LoginPage>
  </Router>
</BrowserRouter>
`;
